relay_client/src/http.rs

```diff
@@ -105,6 +105,7 @@ impl Client {
         &self,
         topic: Topic,
         message: impl Into<Arc<str>>,
+        attestation: impl Into<Option<Arc<str>>>,
```

Comment:
`Option<impl T>` is generally better for a public API, as `Into<Option<A>> for Option<B>` is usually not implemented.

Reply:
My thinking is that if you already have an `Arc<str>`, then you'd be able to omit `Some()`, but in practice that's probably never gonna happen.

I think the conversation to have here is why are we still sticking to `Arc<str>`, even though they likely don't have any significant performance benefits, and only hurt the ergonomics of the APIs.

Reply:
What would be the alternative to `Arc<str>`? I think it's good to have basically free cloning

Reply:
In practice we rarely clone these strings.

The downsides to using `Arc<str>` is double allocation on deserialization (it essentially first deserializes into `String`, and then converts that to `Arc<str>`, if I'm not mistaken) and the ergonomics of working with it (converting `String` to `Arc<str>` is another allocation).

An alternative is using regular `String`. The cloning cost is mostly mitigated by using a fast allocator like jemalloc or mimalloc.

It's hard to say definitively what would be more performant for the relay, but in some cases (a lot of highly multi-threaded cloning) using `String` is a lot faster, since allocators would use essentially thread-local memory arenas with minimal locking, while `Arc<str>` is using atomics under the hood for synchronization. And even though atomic usage is cheap, it's not free.

Reply:
> double allocation on deserialization (it essentially first deserializes into String, and then converts that to Arc<str>, if I'm not mistaken)

Oh wow I didn't know about that. Yeah that looks accurate